tests/eip7702.rs

```diff
@@ -0,0 +1,2720 @@
+
+		// Both authorizations should be processed regardless of order
+		let delegation1 = evm_core::create_delegation_designator(implementation1);
+		let delegation2 = evm_core::create_delegation_designator(implementation2);
+
+		assert_eq!(executor.code(authorizing1), delegation1);
+		assert_eq!(executor.code(authorizing2), delegation2);
+
+		// Both nonces should be incremented
+		assert_eq!(executor.state().basic(authorizing1).nonce, U256::from(1));
+		assert_eq!(executor.state().basic(authorizing2).nonce, U256::from(1));
+	}
+}
+
+// ============================================================================
+// Edge Cases and Security Tests (Section 9)
+// ============================================================================
+
+#[test]
+fn test_9_1_self_delegation() {
+	// Test: EOA delegates to its own address
+	// Expected: Should work but lead to infinite loop prevention or error
+	let caller = H160::from_slice(&[1u8; 20]);
+	let self_delegating = H160::from_slice(&[2u8; 20]);
+	let target = H160::from_slice(&[3u8; 20]);
+
+	let config = Config::pectra();
+	let mut state = BTreeMap::new();
+
+	state.insert(
+		caller,
+		evm::backend::MemoryAccount {
+			nonce: U256::zero(),
+			balance: U256::from(10_000_000),
+			storage: BTreeMap::new(),
+			code: Vec::new(),
+		},
+	);
+
+	state.insert(
+		self_delegating,
+		evm::backend::MemoryAccount {
+			nonce: U256::zero(),
+			balance: U256::from(1000),
+			storage: BTreeMap::new(),
+			code: Vec::new(),
+		},
+	);
+
+	let vicinity = create_test_vicinity();
+	let mut backend = MemoryBackend::new(&vicinity, state);
+
+	let metadata = evm::executor::stack::StackSubstateMetadata::new(1000000, &config);
+	let state = evm::executor::stack::MemoryStackState::new(metadata, &mut backend);
+	let precompiles = BTreeMap::new();
+	let mut executor = StackExecutor::new_with_precompiles(state, &config, &precompiles);
+
+	// Self-delegation: EOA delegates to itself
+	let authorization = create_authorization(
+		U256::from(1),
+		self_delegating, // delegate to self
+		U256::zero(),
+		self_delegating, // authorizing address is same
+	);
+
+	let (exit_reason, _) = executor.transact_call(
+		caller,
+		target,
+		U256::zero(),
+		Vec::new(),
+		100_000,
+		Vec::new(),
+		vec![authorization],
+	);
+
+	assert_eq!(exit_reason, ExitReason::Succeed(evm::ExitSucceed::Stopped));
+
+	// Verify self-delegation was set
+	let delegation_designator = evm_core::create_delegation_designator(self_delegating);
+	assert_eq!(executor.code(self_delegating), delegation_designator);
+
+	// Now try to call the self-delegating address - this should handle the infinite loop
+	let (call_exit_reason, _) = executor.transact_call(
+		caller,
+		self_delegating,
+		U256::zero(),
+		Vec::new(),
+		100_000,
+		Vec::new(),
+		Vec::new(),
+	);
+
+	// Should either succeed (infinite loop prevented) or fail gracefully
+	assert!(
+		matches!(call_exit_reason, ExitReason::Succeed(_))
```

Comment:
It should fail as the EOA can't have valid executable bytecode

Reply:
Addressed in https://github.com/rust-ethereum/evm/pull/323/commits/3b330e24afa839953de526b89c47236c2b37a8e6